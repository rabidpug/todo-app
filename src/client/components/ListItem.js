import { Avatar, List, } from 'antd';

import IconText from './IconText';
import React from 'react';

const { Item, } = List;
const { Meta, } = Item;
const ListItem = ( {
  toggleTodoItem, deleteTodoItem, completedDate, createdDate, text,
} ) => (
  <Item
    actions={ [
      <IconText
        key='markit'
        onClick={ toggleTodoItem }
        text={ completedDate ? 'Mark Incomplete' : 'Mark Completed' }
        type={ completedDate ? 'exclamation-circle-o' : 'check' }
      />,
      <IconText
        key='deleteit'
        onClick={ deleteTodoItem }
        text='Delete'
        type='close-square-o' />,
    ] }
    extra={ <Avatar icon={ completedDate ? 'check' : 'exclamation-circle-o' } /> }>
    <Meta
      description={
        <div>
          {`Created On: ${new Date( createdDate ).toLocaleDateString()} at ${new Date( createdDate ).toLocaleTimeString()}`}
          {completedDate && <br />}
          {completedDate &&
            `Completed On: ${new Date( completedDate ).toLocaleDateString()} at ${new Date( completedDate ).toLocaleTimeString()}`}
        </div>
      }
      title={ text }
    />
  </Item>
);

export default ListItem;
