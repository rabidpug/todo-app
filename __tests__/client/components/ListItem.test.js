import ListItem from 'Components/ListItem';
import React from 'react';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'ListItem', () => {
    it(
      'Should render the ListItem', () => {
        const wrapper = shallow( <ListItem /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
