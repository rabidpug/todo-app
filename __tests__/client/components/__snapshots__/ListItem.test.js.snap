// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Should render the ListItem 1`] = `
<Item
  actions={
    Array [
      <IconText
        onClick={undefined}
        text="Mark Completed"
        type="check"
      />,
      <IconText
        onClick={undefined}
        text="Delete"
        type="close-square-o"
      />,
    ]
  }
  extra={
    <Avatar
      icon="exclamation-circle-o"
      prefixCls="ant-avatar"
      shape="circle"
      size="default"
    />
  }
>
  <Meta
    description={
      <div>
        Created On: Invalid Date at Invalid Date
      </div>
    }
  />
</Item>
`;
